From fe1d84ed29fae5c9e66d42e06579ced1e02eb7f6 Mon Sep 17 00:00:00 2001
From: Chen Wandun <chenwandun@huawei.com>
Date: Mon, 14 Nov 2022 12:16:34 +0800
Subject: [PATCH 0031/1398] [Huawei] mm: disable psi cgroup v1 by default

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187968

-------------------------------

Feature of psi cgroup v1 should only enable when open
CONFIG_PSI_CGROUP_V1.

Signed-off-by: Chen Wandun <chenwandun@huawei.com>
---
 include/linux/psi.h    | 2 +-
 kernel/sched/cpuacct.c | 4 ++--
 kernel/sched/psi.c     | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/linux/psi.h b/include/linux/psi.h
index d290f0493c33..2708dd1194ae 100644
--- a/include/linux/psi.h
+++ b/include/linux/psi.h
@@ -14,7 +14,7 @@ struct css_set;
 extern struct static_key_false psi_disabled;
 extern struct psi_group psi_system;
 
-extern struct static_key_false psi_v1_disabled;
+extern struct static_key_true psi_v1_disabled;
 
 void psi_init(void);
 
diff --git a/kernel/sched/cpuacct.c b/kernel/sched/cpuacct.c
index e51c1f524b8c..aa4617a6f8fe 100644
--- a/kernel/sched/cpuacct.c
+++ b/kernel/sched/cpuacct.c
@@ -383,8 +383,8 @@ static int __init setup_psi_v1(char *str)
 	int ret;
 
 	ret = kstrtobool(str, &psi_v1_enable);
-	if (!psi_v1_enable)
-		static_branch_enable(&psi_v1_disabled);
+	if (psi_v1_enable)
+		static_branch_disable(&psi_v1_disabled);
 
 	return ret == 0;
 }
diff --git a/kernel/sched/psi.c b/kernel/sched/psi.c
index 25f7d46ad7bd..d87a7d810ffc 100644
--- a/kernel/sched/psi.c
+++ b/kernel/sched/psi.c
@@ -156,7 +156,7 @@
 static int psi_bug __read_mostly;
 
 DEFINE_STATIC_KEY_FALSE(psi_disabled);
-DEFINE_STATIC_KEY_FALSE(psi_v1_disabled);
+DEFINE_STATIC_KEY_TRUE(psi_v1_disabled);
 
 #ifdef CONFIG_PSI_DEFAULT_DISABLED
 static bool psi_enable;
-- 
2.17.1

